### DISCUSSION TITLE
Set MATLABPATH when running neurodesktop

### DISCUSSION THREAD

**Discussion by JonteP** [neurodesk/neurodesk.github.io-hugo#412] [Q&A] [answered]
I have a very basic question. I tried to set MATLABPATH when calling docker run to point to a mounted folder. The goal is to have certain scripts/toolboxes available without having to set up the path each time. This did not work, since Matlab will not see my environment variable. Is there a way around this?

**Comment by stebo85** [comment]
Dear @JonteP,
Can you describe in more detail what you are doing and what's not working?

Here is an example how to set the MATLABPATH:

add this to your ~/.bashrc in Neurodesktop:
```
export MATLABPATH=/data/toolboxes
```

Now, when you start Matlab from the Application menu, the variable MATLABPATH is available for matlab:
<img width="674" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/a1339d20-e16b-4e32-80bb-08dfd8d7f982">

**Reply by JonteP** [reply]
I attempted to set it with docker run -e MATLABPATH=/path/to/matlab -v /my/matlab/path:/path/to/matlab so that this could be set with a startup script. Seems like this is only visible to Jupyterlab and not Neurodesktop?

**Reply by stebo85** [reply, marked as answer]
Dear @JonteP 

That works as well. 


however, you need to make sure that the matlab path is visible for the matlab container inside neurodesktop. By default this is everything in the homedirectory and /data

so, this would work:
```
-e MATLABPATH=/data/matlabtoolboxes -v /my/matlab/path/on/my/host/sytem:/data/matlabtoolboxes
```
